admin/atlas_client.go

```diff
@@ -58,24 +60,11 @@ func UseDigestAuth(apiKey, apiSecret string) ClientModifier {
 // credentials.LocalTokenCache can be supplied to reuse OAuth Token across application restarts.
 // Warning: for advanced use cases please use credentials.NewTokenSource directly in your code pass it to UseHTTPClient method.
 // Warning: any previously set httpClient will be overwritten. To fully customize HttpClient use UseHTTPClient method.
-func UseOAuthAuth(clientID, clientSecret string, tokenCache credentials.LocalTokenCache) ClientModifier {
+func UseOAuthAuth(clientID, clientSecret string, tokenCache *oauth2.Token) ClientModifier {
```

Comment:
this should probably accept a context and not sure if other modifiers like the ones for the base url

Reply:
This concept was inherited from old SDK, and it does hold us back due to `UseBaseURL` or `UseHttpClient` overriding effects of this one.

Reply:
I'm not talking about the method but that this method should probably accept a ctx and should maybe allow to override the base url, of the oauth client which right now it doesn't allow me and why I could not really test this

Reply:
UseBaseURL overrides URL global. That is why `&c.Servers[0].URL` were used in the code that was removed.
Context should be also reused across the entire client otherwise we will have very confusing end user API.

Reply:
Adding those changes.

Reply:
>UseBaseURL overrides URL global. That is why &c.Servers[0].URL were used in the code that was removed.

Does that mean I need to know I have to call that function first?

Reply:
Yes. That is why I think we should:
- Deprecate UseDigest 
- Change examples to UseHttpClient instead (for both digest and OAuth)

WDYT?

Reply:
no strong opinion, the CLI already works with `UseHttpClient`

Reply:
I think with this PR we moving from:

- SDK provides simple and comprehensive 
- Just use oauth2 + our solution for token storage we provided.

Reply:
yes, which should've always been the case :)